docker-plugin/src/main/java/com/nirima/jenkins/plugins/docker/utils/WaitForSSHComputerLauncher.java

```diff
@@ -0,0 +1,42 @@
+package com.nirima.jenkins.plugins.docker.utils;
+
+import hudson.model.TaskListener;
+import hudson.plugins.sshslaves.SSHLauncher;
+import hudson.slaves.DelegatingComputerLauncher;
+import hudson.slaves.SlaveComputer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+import static com.nirima.jenkins.plugins.docker.utils.PortUtils.waitForSshOn;
+
+/**
+ * @author mvtorres, lanwen (Merkushev Kirill)
+ *
+ * Waits for ssh connection can be esabilished
+ *
+ * throws IllegalStateException if port is not connectible
+ * throws IOException if it can't connect to ssh daemon
+ */
+public class WaitForSSHComputerLauncher extends DelegatingComputerLauncher {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WaitForSSHComputerLauncher.class);
+
+    private String host;
+    private int port;
+
+    public WaitForSSHComputerLauncher(SSHLauncher delegate) {
+        super(delegate);
+        this.host = delegate.getHost();
+        this.port = delegate.getPort();
+    }
+
+    @Override
+    public void launch(SlaveComputer computer, TaskListener listener) throws IOException, InterruptedException {
+        LOGGER.info("Try to connect to sshd on {}:{}", host, port);
+
+        waitForSshOn(host, port);
+        super.launch(computer, listener);
```

Comment:
I think after #234 it will be possible just to have retries as additional option for DockerSSHComputerLauncher, because it already wrapping sshlauncher/sshconnector to have ability to choose/configure launcher in UI.